src/main/java/frc/robot/hardware/mechanisms/FlywheelSimulation.java

```diff
@@ -0,0 +1,48 @@
+package frc.robot.hardware.mechanisms;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+import frc.utils.time.TimeUtils;
+
+public class FlywheelSimulation extends MechanismSimulation {
+
+	private final FlywheelSim flywheelSimulation;
+
+	private Rotation2d position;
+	private Rotation2d lastVelocity;
+	private Rotation2d currentVelocity;
+
+	public FlywheelSimulation(FlywheelSim flywheelSimulation, double gearRatio) {
+		super(gearRatio);
+		this.flywheelSimulation = flywheelSimulation;
+		this.position = Rotation2d.fromDegrees(0);
+		this.lastVelocity = getSystemVelocityAnglesPerSecond();
+		this.currentVelocity = getSystemVelocityAnglesPerSecond();
+	}
+
+	@Override
+	public Rotation2d getSystemPosition() {
+		return position;
+	}
+
+	@Override
+	public Rotation2d getSystemVelocityAnglesPerSecond() {
+		return Rotation2d.fromRadians(flywheelSimulation.getAngularVelocityRadPerSec());
+	}
+
+	@Override
+	public void setInputVoltage(double voltage) {
+		flywheelSimulation.setInputVoltage(voltage);
+	}
+
+	@Override
+	public void updateMotor() {
+		flywheelSimulation.update(TimeUtils.getCurrentCycleTimeSeconds());
+		currentVelocity = getSystemVelocityAnglesPerSecond();
+		Rotation2d distanceDelta = Rotation2d.fromRotations(((currentVelocity.getRotations() + lastVelocity.getRotations()) / 2))
```

Comment:
```suggestion
		Rotation2d deltaDistance = Rotation2d.fromRotations(((currentVelocity.getRotations() + lastVelocity.getRotations()) / 2))
```

Reply:
I'm not sure, it's the delta of the distance not distance of the delta

Reply:
but that's a math thing, its delta before the thing:
delta x, delta y, delta v

Reply:
Δx, Δy, Δv

Reply:
but you are writing it in English not mathematical language

Reply:
yea but still.
I could להתפשר on using "rotationDifference" times "deltaTime" = "distanceSinceLastCycle"

Reply:
but deltaTime stays deltaTime

Reply:
average velocity

Reply:
but still you need the variable for distance delta

Reply:
distanceSinceLastCycle

Reply:
@emilyyyylime @Sh-Pe @YHerm 

distanceDelta or deltaDistance?

Reply:
I mean I guess it is more correct to write deltaDistance because it's the standard way. If wpilib were to write this class I'm sure they would write dDistance or dx. But if I were to name this I would write distanceDifference because it is the difference of the distance